Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1977,6 +1979,25 @@ PhysicalParticleContainer::getIonizationFunc (const WarpXParIter& pti,
                                 ion_atomic_number);
 }
 
+void PhysicalParticleContainer::resample (const Resampling& resampler, const int timestep)
+{
+    const amrex::Real global_numparts = TotalNumberOfParticles();
+
+    if (resampler.triggered(timestep, global_numparts))
+    {
+        auto & warpx = WarpX::GetInstance();
+        for (int lev = 0; lev <= warpx.maxLevel(); lev++)
```

Comment:
`amrex::ParticleContainer` already has a `maxLevel` function, which should return the same thing, so this might work.
```suggestion
        for (int lev = 0; lev <= maxLevel(); lev++)
```

Reply:
It does work, thanks.